From 0d1a25e927850edcc6f777523b850c25cc1ef20b Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 17 Jul 2024 09:25:16 +0300
Subject: [PATCH] [FSW-2704] fs1: create enum for rom manifest type

From rom perspective, it is only primary or secondary manifest.
This patch adds enum to define these types and changes in
functions where this enum is required.

Change-Id: Ia9ffe946a01e08b7efd9b53ba813a3b7eae318fc
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 7603573..3c39aac 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -65,8 +65,8 @@
 	boot_rom_boot_flow_wait_for_dma_finish();
 
 	/* If both manifests are invalid enter into recovery */
-	if (!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_0) &&
-		!boot_rom_boot_flow_is_manifest_valid(HL_FW_SPI_MANIFEST_1)) {
+	if (!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_PRIMARY) &&
+		!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_SECONDARY)) {
 		execute_recovery();
 	}
 
diff --git a/common/boot_flow.c b/common/boot_flow.c
index cd8186d..97cc8af 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -135,12 +135,12 @@
 		secondary_manifest_valid = true;
 }
 
-bool boot_rom_boot_flow_is_manifest_valid(enum hl_fw_manifest_type man_idx)
+bool boot_rom_boot_flow_is_manifest_valid(enum boot_rom_manifests rom_manifest_type)
 {
-	if (man_idx == primary_manifest_idx)
-		return primary_manifest_valid;
-	else
+	if (rom_manifest_type)
 		return secondary_manifest_valid;
+	else
+		return primary_manifest_valid;
 }
 
 bool boot_rom_boot_flow_is_secondary_manifest_valid(void)
diff --git a/common/boot_flow.h b/common/boot_flow.h
index 52c05fb..338f502 100644
--- a/common/boot_flow.h
+++ b/common/boot_flow.h
@@ -9,7 +9,7 @@
 uint32_t boot_rom_boot_flow_prefetch_img(enum hl_fw_img_type img_type);
 void boot_rom_boot_flow_boot_ppboot(enum hl_fw_manifest_type man_idx);
 void boot_rom_boot_flow_validate_manifests(void);
-bool boot_rom_boot_flow_is_manifest_valid(enum hl_fw_manifest_type man_idx);
+bool boot_rom_boot_flow_is_manifest_valid(enum boot_rom_manifests rom_manifest_type);
 bool boot_rom_boot_flow_is_secondary_manifest_valid(void);
 void boot_rom_boot_flow_wait_for_dma_finish(void);
 
